docs/build/building-modules/15-depinject.md

```diff
@@ -104,18 +104,28 @@ All methods, structs and their fields must be public for `depinject`.
 
 5. Create a function named `ProvideModule` (as called in 1.) and use the inputs for instantiating the module outputs.
 
-  ```go reference
-  https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-alpha.0/x/group/module/module.go#L220-L235
-  ```
+    ```go reference
+    https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-alpha.0/x/group/module/module.go#L220-L235
+    ```
+
+    The `ProvideModule` function should return an instance of `cosmossdk.io/core/appmodule.AppModule` which implements
+    one or more app module extension interfaces for initializing the module.
+
+    Following is the complete app wiring configuration for `group`:
+
+    ```go reference
+    https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-alpha.0/x/group/module/module.go#L194-L235
+    ```
+
+6. All modules must implement `depinject.OnePerModuleType` interface. This is used in order to tell the dependency injection framework that the module can only be instaniated once. 
```

Comment:
Nice, can we add `_ depinject.OnePerModuleType = AppModule{}` in all our modules, close to our type checks?

https://github.com/cosmos/cosmos-sdk/blob/v0.50.2/x/gov/module.go#L41-L50